modules/jaeger-config-storage.adoc

```diff
@@ -0,0 +1,207 @@
+Minimum deployment = 16Gi*
+|500m
+
+|elasticsearch:
+  resources:
+    limits:
+      cpu:
+|Limit on number of central processing units, based on your environment’s configuration.
+|Specified in cores or millicores (for example, 200m, 0.5, 1). For example, Proof of concept = 500m,
+Minimum deployment =1
+|
+
+|elasticsearch:
+  resources:
+    limits:
+      memory:
+|Available memory limit based on your environment’s configuration.
+|Specified in bytes (for example, 200Ki, 50Mi, 5Gi). For example, Proof of concept = 1Gi,
+Minimum deployment = 16Gi*
+|
+
+|elasticsearch:
+  secretname:
+|Name of the secret, for example `jaeger-secret`.
+|
+|N/A
+
+
+|storage:
+  esIndexCleaner:
+    enabled:
+|When using Elasticsearch storage, by default a job is created to clean old traces from the index.  This parameter enables or disables the index cleaner job.
+|`true`/ `false`
+|`true`
+
+|storage:
+ esIndexCleaner:
+  numberOfDays:
+|Number of days to wait before deleting an index.
+|Integer value
+|`7`
+
+|storage:
+ esIndexCleaner:
+  schedule:
+|Defines the schedule for how often to clean the Elasticsearch index.
+|Cron expression
+|"55 23 * * *"
+
+|storage:
+ esRollover:
+  schedule:
+|Defines the schedule for how often to rollover to a new Elasticsearch index.
+|Cron expression
+|'*/30 * * * *'
+|
+3+|{asterisk} Each Elasticsearch node can operate with a lower memory setting though this is *not* recommended for production deployments. For production use, you should have no less than 16Gi allocated to each Pod by default, but preferably allocate as much as you can, up to 64Gi per Pod.
+|===
+
+.Production storage example
+[source,yaml]
+----
+apiVersion: jaegertracing.io/v1
+kind: Jaeger
+metadata:
+  name: simple-prod
+spec:
+  strategy: production
+  storage:
+    type: elasticsearch
+    elasticsearch:
+      nodeCount: 1
+      resources:
+        requests:
+          cpu: 500m
+          memory: 1Gi
+        limits:
+          memory: 1Gi
+----
+
+
+.Storage example with volume mounts
+[source,yaml]
+----
+apiVersion: jaegertracing.io/v1
+kind: Jaeger
+metadata:
+  name: simple-prod
+spec:
+  strategy: production
+  storage:
+    type: elasticsearch
+    options:
+      es:
+        server-urls: https://quickstart-es-http.default.svc:9200
+        index-prefix: my-prefix
+        tls:
+          ca: /es/certificates/ca.crt
+    secretName: jaeger-secret
+  volumeMounts:
+    - name: certificates
+      mountPath: /es/certificates/
+      readOnly: true
+  volumes:
+    - name: certificates
+      secret:
+        secretName: quickstart-es-http-certs-public
+----
+
+
+.Jaeger example with a single node Elasticsearch cluster with AWS gp2 persistent storage:
+[source,yaml]
+----
+apiVersion: jaegertracing.io/v1
+kind: Jaeger
+metadata:
+  name: simple-prod
+spec:
+  strategy: production
+  storage:
+    type: elasticsearch
+    elasticsearch:
+      nodeCount: 1
+      storage:
+        storageClassName: gp2
```

Comment:
Describe how storage provisioning works. For instance how it can be reused. It is documented in the upstream docs in the operator/ES section.

Reply:
Will cover this in a subsequent PR.